app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -308,7 +308,7 @@ def create_vm(args = {})
     end
 
     def get_ovirt_id(argument_list, argument)
-      if argument_list.detect { |a| a.name == argument }.nil? && argument_list.detect { |a| a.id == argument }.nil?
+      if !argument.nil? && argument_list.detect { |a| a.name == argument }.nil? && argument_list.detect { |a| a.id == argument }.nil?
```

Comment:
Is a `nil` argument actually valid? or should that just throw a different error, such as `missing id for ...`?
also, how about 
```suggestion
      if !argument.nil? && argument_list.none? { |a| a.name == argument || a.id == argument }
```